src/backend/cuda/subconfigure.m4

```diff
@@ -57,28 +57,36 @@ AC_ARG_WITH([cuda-sm],
 
 
 # --with-cuda
+AC_ARG_VAR([NVCC], [nvcc compiler to use])
 PAC_SET_HEADER_LIB_PATH([cuda])
 if test "$with_cuda" != "no" ; then
     PAC_CHECK_HEADER_LIB([cuda_runtime_api.h],[cudart],[cudaStreamSynchronize],[have_cuda=yes],[have_cuda=no])
     if test "${have_cuda}" = "yes" ; then
-        AC_MSG_CHECKING([whether nvcc works])
-        if test -n "$ac_save_CC" ; then
-            NVCC_FLAGS="$NVCC_FLAGS -ccbin $ac_save_CC"
-            # - pgcc/nvc doesn't work, use pgc++/nvc++ instead
-            # - Extra optins such as `gcc -std=gnu99` doesn't work, strip the option
-            NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/nvc/nvc++/g' -e 's/pgcc/pgc++/g' -e's/ -std=.*//g')
-        fi
-        # try nvcc from PATH if 'with-cuda' does not contain a valid path
-        if test -d ${with_cuda} ; then
-            nvcc_bin=${with_cuda}/bin/nvcc
+        # check if NVCC has been set by the user
+        if test -z "$NVCC" ; then
+            if test -n "$ac_save_CC" ; then
+                NVCC_FLAGS="$NVCC_FLAGS -ccbin $ac_save_CC"
+                # - pgcc/nvc doesn't work, use pgc++/nvc++ instead
+                # - Extra optins such as `gcc -std=gnu99` doesn't work, strip the option
+                NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/nvc/nvc++/g' -e 's/pgcc/pgc++/g' -e's/ -std=.*//g')
+            fi
+
+            # try nvcc from PATH if 'with-cuda' does not contain a valid path
+            if test -d ${with_cuda} ; then
+                nvcc_bin=${with_cuda}/bin/nvcc
+            else
+                nvcc_bin=nvcc
+            fi
         else
-            nvcc_bin=nvcc
+            nvcc_bin=$NVCC
+            AC_MSG_WARN([Using user-provided nvcc: '${nvcc_bin}'])
         fi
 
         # save language settings, customize ac_ext and ac_compile to support CUDA
         AC_LANG_PUSH([C])
         ac_ext=cu
         ac_compile='$nvcc_bin $NVCC_FLAGS -c conftest.$ac_ext >&5'
```

Comment:
Sorry for the late response. How about make it `ac_compile="$NVCC -c ...`, i.e. `NVCC="$nvcc_bin $NVCC_FLAGS"`? The intention is to allow user fully overwrite `NVCC`.

I am surprised to see single quotes here. Wouldn't the single quote prevent shell variable expansion? What did I miss?

Reply:
> > I am surprised to see single quotes here. Wouldn't the single quote prevent shell variable expansion? What did I miss?
> 
> Mh good point, oversaw that but it is working as expected, e.g., I saw something like `configure:18564: WARNING: Using user-provided nvcc: 'nvcc -ccbin gcc'` when testing. I did not take a closer look at the autoconf macros but would assume they expand, don't they?

That warning is from line 82, where the single quote is inside a double-quoted string.
Line 88 is not an autoconf macro. It's normal shell variables. Since it works, I am probably missed something here. I'll follow up.

Reply:
> Line 88 is not an autoconf macro. It's normal shell variables. Since it works, I am probably missed something here. I'll follow up.

Yes, I know. But `ac_compile` is certainly used (and then expanded) by an autoconf macro I would have assumed.

Reply:
Fixed this and also changed the behavior w.r.t. `NVCC` and `NVCC_FLAGS`.